mockwebserver/src/main/kotlin/mockwebserver3/MockWebServer.kt

```diff
@@ -855,31 +849,30 @@ public class MockWebServer : Closeable {
       // Even if messages are no longer being read we need to wait for the connection close signal.
       connectionClose.await()
     } finally {
-      source.closeQuietly()
+      socket.source.closeQuietly()
```

Comment:
Can this just be socket.closeQuietly() now?

Reply:
or is this relying on

>  Note that [asOkioSocket] returns a socket that closes the underlying [java.net.Socket] when both of its component streams are closed.

Reply:
Yep. I’m working towards that but I’m not there yet.